lib/MusicBrainz/Server/Data/WebService.pm

```diff
@@ -204,21 +204,24 @@ sub xml_search
     }
 
     my $url_ext;
+    my $redirect_url;
     if (DBDefs->SEARCH_ENGINE eq 'LUCENE') {
         my $format = ($args->{fmt} // "") eq "json" ? "jsonnew" : "xml";
+        $redirect_url = '/internal/search/'
         $url_ext = "/ws/2/$resource/?" .
            "max=$limit&type=$resource&fmt=$format&offset=$offset" .
            "&query=" . uri_escape_utf8($query) . "&dismax=$dismax";
     } else {
         my $format = ($args->{fmt} // "") eq "json" ? "mbjson" : "mbxml";
         my $def_type = $dismax eq 'true' ? 'dismax' : 'lucene';
+        $redirect_url = '/internal/solr/'
         $url_ext = "/$resource/select?" .
             "rows=$limit&wt=$format&start=$offset" .
             "&q=" . uri_escape_utf8($query) . "&defType=$def_type";
     }
 
     if (DBDefs->SEARCH_X_ACCEL_REDIRECT) {
```

Comment:
Since the trick is only related to X_ACCEL thing, i would keep the code together:
```perl
my $redirect_url = (DBDefs->SEARCH_ENGINE eq 'LUCENE') ? '/internal/search/' : '/internal/solr/';
return { redirect_url => $redirect_url . DBDefs->SEARCH_SERVER . $url_ext }
```

Reply:
It also unbreaks the code (missing semicolons :)).